### ISSUE TITLE
Add the new forge

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#557]
- [ ] Readd the forge as a room that can be added in game now the workshop as taken its place.
- [ ] Add the room in the research tree.
- [ ] Add the creature equipment crafting logic.

**Comment by hwoarangmy** [comment]
Well, now that you have removed the forge, adding an empty room that will be disabled might not be that relevant (and waste of time). IMHO, that's better to stay as it is and we will see for 0.6.0. We have more important stuff than a disabled room to focus on for now.

**Comment by Bertram25** [comment]
> Well, now that you have removed the forge, adding an empty room that will be disabled might not be that relevant (and waste of time). IMHO, that's better to stay as it is and we will see for 0.6.0. We have more important stuff than a disabled room to focus on for now.

Yeah, well, you were the one asking to readd the forge that's why I did it that way. This can be fully moved to 0.6.0, I don't mind at all.

**Comment by hwoarangmy** [comment]
> you were the one asking to readd the forge that's whay I did it that way

I was just asking why you removed it, not asking to re-add it ;)
But now that it's done, no need to waste time for that.

**Comment by Bertram25** [comment]
> I was just asking why you removed it, not asking to re-add it ;)
> But now that it's done, no need to waste time for that.

Ah ok. :) Let's move that out then.

**Comment by hwoarangmy** [comment]
Just to clarify, how do you see the forge work ?
- Which creatures will forge weapons ? Each one for itself (might be a good idea since it would make the creature use some time it instead of training) ? Or some creatures good at forging ?
- If the creatures do not craft their own weapons, how do we decide which weapon should be crafted ? Depending on the creatures in the dungeon ? Randomly ?
- We said that some rogue creatures might drop weapons on death. How will it deal with the forge concept ?
- In the end, I guess we will have several weapon levels (for example basic sword, good sword and heaven sword). What will limit the weapon power of the weapons a creature will forge ?

@OpenDungeons/developers WDYT ?

**Comment by Danimal696** [comment]
> In the end, I guess we will have several weapon levels (for example basic sword, good sword and heaven sword). What will limit the weapon power of the weapons a creature will forge ?

I would say a research topic for each weapon level crafting

**Comment by hwoarangmy** [comment]
> I would say a research topic for each weapon level crafting

Ok, sounds fair. However, since you have thought a bit more than me about forge, how do you see it concerning the other questions ?

**Comment by Danimal696** [comment]
> Which creatures will forge weapons ? Each one for itself (might be a good idea since it would make the creature use some time it instead of training) ? Or some creatures good at forging ?

Either another use for goblins/smiths (complex choice) or each creatures forges their own weapon in forge (simple choice), goblin would forge them and weapons would be stored in another room (armory) waiting for the taking

> If the creatures do not craft their own weapons, how do we decide which weapon should be crafted ? Depending on the creatures in the dungeon ? Randomly ?

By crafting generic weapons? 5 good weapons would mean 5 weapons of good quality for the taking regardless of kind, any upgrade-able creature could go there and consume 1 unit (not caring if its a crossbow or an axe). Goblin/smiths work in forge on their spare time, with workshops having top priority to craft traps. Maybe give one creature a strong like to weapon crafting?, the planned advanced smith creatures could prefer weapon crafting to trap crafting (Runelord for heroes, ??? for keepers). I think this implies a slower game pace more fit for the campaign.

> We said that some rogue creatures might drop weapons on death. How will it deal with the forge concept ?

Just adding a +1 to the stock of that weapons quality in an armory (there could even a % chance that an equipped creature drops their weapon at death, something like 15%, making it cannon killed creatures have their weapons destroyed in combat most of the time to excuse this device).

Armory room: each 3x3 tiles houses an active spot that can hold only 1 weapon of each quality regardless of researched or not. (think of a weapon rack with 3 shelves, the shelf gets full when a weapon is stored in there, lower for good, middle for magic, top for epic qualities) OR function like treasure room, and production on forge generates weapon points that are stored on armories and spent depending on creature demand, being the researches the one thing limiting their quality demand. I prefer the first option, now, how should creatures take weapons? by dropping in armory or letting them check and try to be the first to armory when new toys are ready?

**Comment by hwoarangmy** [comment]
> and weapons would be stored in another room (armory) waiting for the taking

That would mean the need for 2 rooms for 1 purpose. Which also means 2 researches (1 alone being useless). IMHO, it's a little bit too complex for a concept pretty simple. Moreover, we are lacking rooms and using 2 for that may be a little too much. However, you are the one making rooms so...
Note that as a first step, if we agree on how the forge should behave, I had thought about taking 1 of the models of the workshop and use it for the armory (and keep the same wall active spots for both rooms). It will be a loss of diversity but we would have 2 rooms available without needing more models. Finding a ground texture is easier than the models.

> By crafting generic weapons?

Ok. But that would mean to have, for each weapon, a fixed number of values. That wmay require many models... But it's up to you.

> Just adding a +1 to the stock of that weapons quality in an armory (there could even a % chance that an equipped creature drops their weapon at death, something like 15%, making it cannon killed creatures have their weapons destroyed in combat most of the time to excuse this device).

The good thing with your idea of dropping "generic" weapons is that it would make weapon dropping pretty easy. And dropping weapon of creatures handling them too. Even if it might look weird to have a knight with a sword drop a weapon and use it for a bow.

> how should creatures take weapons? by dropping in armory or letting them check and try to be the first to armory when new toys are ready?

IMHO, when we will be clear and have a good vision on what we want for the forge/armory, the answer to that will be pretty easy.

@OpenDungeons/developers WDYT ?

**Comment by Danimal696** [comment]
> - By crafting generic weapons?
> 
> Ok. But that would mean to have, for each weapon, a fixed number of values. That wmay require many models... But it's up to you.

I meant, by creating a weapon point that can be spent on any weapon of its quality.

Armory models would be pretty easy to make, since they are mostly ready by using the very same models of weapons creatures will equip. The forge research could unlock both rooms, OR forge could store the weapons on its wall spots and center ones would be the ones for working. Total walls spot would be the ones to determinate the total weapon storage of the room (3 wall spots = 3 good, 3 magic, 3 epic weapons storage) no need to have a graphical representation of each weapon, we just need to show it full even if there is only one weapon there (0 weapons = empty racks)
http://opengameart.org/content/wooden-weapon-training-set
http://opengameart.org/content/undead-weapon-set-and-weapon-rack

Chimmey model would have to be discarded, no loss here.

> Even if it might look weird to have a knight with a sword drop a weapon and use it for a bow.

Do you plan for creatures to take weapons on the field? dropped weapons could be seen as "damaged weapons" so a kobold needs to take it to a forge storage before being used (like if it had to be inspected/repaired before handling it down). It would be one way to solve that  "sword to crossbow" problem.

**Comment by hwoarangmy** [comment]
IMHO, using 1 room is better to keep the same behaviour as the other ones. The game pace is already slow, it would be better to not slow it even more.
However, we always said we wanted to have different "paths". Having good weapons might be a good way to achieve that so that one can choose to upgrade his creatures or try to equip them so that they can handle upgraded creatures.

However, we should not use wall active spots to keep weapons IMHO. That's because it would mean that a forge without wall active spots would be useless and that's not how all the other rooms work.

> Do you plan for creatures to take weapons on the field? dropped weapons could be seen as "damaged weapons" so a kobold needs to take it to a forge storage before being used (like if it had to be inspected/repaired before handling it down). It would be one way to solve that "sword to crossbow" problem.

Yes, bringing weapons to the forge to get points could be a way to solve the problem. However, we won't have the possibility to have some creatures drop unique weapons.
